### PR TITLE
Included SQL to return unique constraints for Sybase

### PR DESCRIPTION
**Pull request by crenan**
## Impact

- [x] Bug fix (non-breaking change which fixes expected existing functionality)
- [ ] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Description

- Adjusted return SQL for Sybase unique constraints used in snapshot.
- Formated [SybaseDatabaseTest.groovy](https://github.com/liquibase/liquibase/compare/master...crenan:liquibase:patch-1?expand=1#diff-a80ad968c122d9182db50ea7a646bb0e7634bc9ad44f864a7564c3abd8e1eb5d)
- Included [SybaseDatabaseTest.java](https://github.com/liquibase/liquibase/compare/master...crenan:liquibase:patch-1?expand=1#diff-b4402f783a4fb313f21915780a3df6b693ca0beedad22257b61b009ac079c3cb) for basic tests over SybaseDatabase class.

## Things to be aware of

- Followed the documentation on [SAP Infocenter](https://infocenter.sybase.com/help/index.jsp?topic=/com.sybase.infocenter.dc36274.1570/html/tables/tables6.htm) (version 16.0 but works for all versions of SAP Sybase ASE).

## Things to worry about

- I didn't found tests nor did I created new ones to cover this change.

## Additional Context

- I found the TODO when I tried to use db-doc command in CLI Liquibase.
- The new URL prefix is for use with jconn4.jar (JDBC driver provided by SAP for Sybase ASE).

### CODE REVIEW

**Review by MalloD12 — APPROVED** [review]
Approved.

# Review and testing results:

Overall this changes looks ok to me:

- Made a bit of refactoring by moving JUnit tests to groovy format 
- Removed JUnit tests class
- Build and test run successfully

### Things to be aware of:
- Nothing else more than what has been described in this PR.

### Things to worry about:
- Nothing else more than what has been described in this PR.

**Review by filipelautert — APPROVED** [review]
Sysbase only change with tests, thanks!

**Review by suryaaki2 — APPROVED** [review]

File: liquibase-core/src/test/groovy/liquibase/database/core/SybaseDatabaseTest.groovy [hunk 1]
```diff
@@ -3,83 +3,111 @@ package liquibase.database.core
...
     @Unroll
-	def getViewDefinition() throws Exception {
+    def getViewDefinition() throws Exception {
         when:
-        def executor = Mock(Executor)
-        executor.queryForList(_ as GetViewDefinitionStatement, String.class) >> viewRows
-		SybaseDatabase database = new SybaseDatabase()
-		Scope.currentScope.getSingleton(ExecutorService.class).setExecutor("jdbc", database, executor)
+            def executor = Mock(Executor)
+            executor.queryForList(_ as GetViewDefinitionStatement, String.class) >> viewRows
```
Comment (XDelphiGrl) [inline liquibase-core/src/test/groovy/liquibase/database/core/SybaseDatabaseTest.groovy:34]: This PR taught me that "_ is a special character in Spock and it means any , it is used as an argument matcher, in this case it is basically ensuring the argument  is indeed GetViewDefinitionStatement class."

Thank you, @suryaaki2 !

**Review by XDelphiGrl — APPROVED** [review]
This fix adds the ability for Liquibase to snapshot unique constraints for the Sybase database platform. 
* New integration tests added for Sybase.
* Functional and test harness executions passing.
* No additional testing required.

APPROVED

**Comment by crenan** [comment]
@filipelautert, this PR was not merged correctly. I updated Liquibase to version 4.19 and I'm getting the error:

```sql
Unexpected error running Liquibase: Error executing SQL select CONSTRAINT_NAME, COLUMN_LIST as COLUMN_NAME, constraint_schema as CONSTRAINT_CONTAINER from information_schema.constraints where constraint_type='UNIQUE' and constraint_catalog='XXXXX' and constraint_schema='dbo' and table_name='XXXXX' and constraint_name='XXXXX': information_schema.constraints not found. Specify owner.objectname or use sp_help to check whether the object exists (sp_help may produce lots of output).
```
_I changed the object names to XXXXX for privecy._

So I went to check the file [JdbcDatabaseSnapshot.java](https://github.com/liquibase/liquibase/pull/3517/files#diff-de7ddcd632d7bb0e4ce3aa1d725f8870e81a37ecf43758a68d08e85d8662ca31) and the changes were gone, event the `} else if (database instanceof SybaseDatabase) {` line is not there.

Do I need to open a new PR?

**Comment by MalloD12** [comment]
> @filipelautert, this PR was not merged correctly. I updated Liquibase to version 4.19 and I'm getting the error:
> 
> ```sql
> Unexpected error running Liquibase: Error executing SQL select CONSTRAINT_NAME, COLUMN_LIST as COLUMN_NAME, constraint_schema as CONSTRAINT_CONTAINER from information_schema.constraints where constraint_type='UNIQUE' and constraint_catalog='XXXXX' and constraint_schema='dbo' and table_name='XXXXX' and constraint_name='XXXXX': information_schema.constraints not found. Specify owner.objectname or use sp_help to check whether the object exists (sp_help may produce lots of output).
> ```
> 
> _I changed the object names to XXXXX for privacy._
> 
> So I went to check the file [JdbcDatabaseSnapshot.java](https://github.com/liquibase/liquibase/pull/3517/files#diff-de7ddcd632d7bb0e4ce3aa1d725f8870e81a37ecf43758a68d08e85d8662ca31) and the changes were gone, event the `} else if (database instanceof SybaseDatabase) {` line is not there.
> 
> Do I need to open a new PR?

Hi @crenan, I looked at this in **master**. And it seems to be in the same way it was originally in the PR. Can you please look at this PR and confirm whether this is what you are expecting for this case or not?
Screenshot 2023-02-06 at 11 20 03 AM

Thanks,
Daniel.

**Comment by crenan** [comment]
Hi @MalloD12.

Sorry, my mistake.
I found that the error I mentioned occurs in [UniqueConstraintSnapshotGenerator](https://github.com/liquibase/liquibase/blob/master/liquibase-core/src/main/java/liquibase/snapshot/jvm/UniqueConstraintSnapshotGenerator.java) file.
In this file, there is a custom SQL only for Sybase ASA. As I'm using Sybase ASE, the code falls in the general handler (line 375).

I'll try to fix this error as soon as possible in another PR.
Best regards.

**Comment by MalloD12** [comment]
> Hi @MalloD12.
> 
> Sorry, my mistake. I found that the error I mentioned occurs in [UniqueConstraintSnapshotGenerator](https://github.com/liquibase/liquibase/blob/master/liquibase-core/src/main/java/liquibase/snapshot/jvm/UniqueConstraintSnapshotGenerator.java) file. In this file, there is a custom SQL only for Sybase ASA. As I'm using Sybase ASE, the code falls in the general handler (line 375).
> 
> I'll try to fix this error as soon as possible in another PR. Best regards.

No worries, thank you!

**Comment by filipelautert** [comment]
Thanks @crenan , once you have the new PR ready let us know and we will try to handle it quickly to get into the next release.

**Comment by crenan** [comment]
> Thanks @crenan , once you have the new PR ready let us know and we will try to handle it quickly to get into the next release.

Hi @filipelautert. Here is the PR for the fix (https://github.com/liquibase/liquibase/pull/3911).
Sorry for taking so long.

**Comment by filipelautert** [comment]
Thanks!

**Pull request merged**